alarm_active_table.cpp

```diff
@@ -307,32 +308,36 @@ alarmActiveTable_SNMPDateTime alarm_time_issued(void)
  */
 void alarmActiveTable_trap_handler(AlarmTableDef& def)
 {
-  alarmActiveTable_context* existing_row = alarm_indexes_to_rows[def.alarm_index()];
-  // 1) New alarm
-  if (existing_row == NULL && def.severity() != AlarmDef::CLEARED)
-  {
-    alarmActiveTable_create_row(def); 
-  }
-  // 2) Existing alarm with different severity
-  else if (existing_row != NULL && def.severity() != AlarmDef::CLEARED && def.severity() != existing_row->_alarm_table_def->severity())
-  {
-    alarmActiveTable_delete_row(def);
-    alarmActiveTable_create_row(def);
-  }
-  // 3) Existing alarm with same severity
-  else if (existing_row != NULL && def.severity() != AlarmDef::CLEARED && def.severity() == existing_row->_alarm_table_def->severity())
-  {
-    // No-op
-  }
-  // 4) Clearing a raised alarm
-  else if (existing_row != NULL && def.severity() == AlarmDef::CLEARED)
-  {
-    alarmActiveTable_delete_row(def);
-  }
-  // 5) Clearing an unraised alarm 
-  else if (existing_row == NULL && def.severity() == AlarmDef::CLEARED)
+  // This stops UTs from failing due to mocking up the cb object
```

Comment:
I might say "from failing when the alarm table is uninitialized" rather than "from failing due to mocking up the cb object" - I think that's a bit clearer to future readers.